File: leetcode/easy/two-sum.py
-- Two Sum
-- https://leetcode.com/problems/two-sum
-- difficulty: easy
-- first_seen: 2026-07-08 03:23:37 EDT

/*
Notes:

*/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        j = len(nums)
        for i in range(j):
            next = target - nums[i]
            if next in numMap:
                return [numMap[next], i]
            numMap[nums[i]] = i
        return []

File: leetcode/easy/two-sum.cpp
// Two Sum
// https://leetcode.com/problems/two-sum
// difficulty: easy
// first_seen: 2026-08-13 23:09:02 EDT
// runtime: 0ms

/*
Notes:

*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        int n = nums.size();

        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];
            if (numMap.count(complement)) {
                return {numMap[complement], i};
            }
            numMap[nums[i]] = i;
        }

        return {}; // No solution found
    }
};